comp/core/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -69,15 +72,18 @@ type resourceTags struct {
 func NewCollector(deps dependencies) (workloadmeta.CollectorProvider, error) {
 	return workloadmeta.CollectorProvider{
 		Collector: &collector{
-			id:                    collectorID,
-			resourceTags:          make(map[string]resourceTags),
-			seen:                  make(map[workloadmeta.EntityID]struct{}),
-			catalog:               workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
-			config:                deps.Config,
-			taskCollectionEnabled: util.IsTaskCollectionEnabled(deps.Config),
-			taskCache:             cache.New(deps.Config.GetDuration("ecs_task_cache_ttl"), 30*time.Second),
-			taskRateRPS:           deps.Config.GetInt("ecs_task_collection_rate"),
-			taskRateBurst:         deps.Config.GetInt("ecs_task_collection_burst"),
+			id:                           collectorID,
+			resourceTags:                 make(map[string]resourceTags),
+			seen:                         make(map[workloadmeta.EntityID]struct{}),
+			catalog:                      workloadmeta.NodeAgent | workloadmeta.ProcessAgent,
+			config:                       deps.Config,
+			taskCollectionEnabled:        util.IsTaskCollectionEnabled(deps.Config),
+			taskCache:                    cache.New(deps.Config.GetDuration("ecs_task_cache_ttl"), 30*time.Second),
+			taskRateRPS:                  deps.Config.GetInt("ecs_task_collection_rate"),
+			taskRateBurst:                deps.Config.GetInt("ecs_task_collection_burst"),
+			metadataRetryInitialInterval: deps.Config.GetDuration("ecs_metadata_retry_initial_interval"),
+			metadataRetryMaxElapsedTime:  deps.Config.GetDuration("ecs_metadata_retry_max_elapsed_time"),
+			metadataRetryTimeoutFactor:   deps.Config.GetInt("ecs_metadata_retry_timeout_factor"),
```

Comment:
If those new parameters are only used to create the ECS client and can never be modified, could we directly create the client here instead of at each `getTaskWithTagsFromV4Endpoint` invocation ?

```suggestion
			client: v3or4.NewClient(metaURI, "v4", v3or4.WithTryOption(
				deps.Config.GetDuration("ecs_metadata_retry_initial_interval"),
				deps.Config.GetDuration("ecs_metadata_retry_max_elapsed_time"),
				func(d time.Duration) time.Duration { return time.Duration(deps.Config.GetInt("ecs_metadata_retry_timeout_factor")) * d }),
			)
```